app/gui2/src/composables/stackNavigator.ts

```diff
@@ -30,7 +30,10 @@ export function useStackNavigator() {
   })
 
   function stackItemToLabel(item: StackItem): string {
-    return graphStore.db.stackItemToMethodName(item) ?? 'unknown'
+    const methodName = graphStore.db.stackItemToMethodName(item)
+    if (!methodName) return 'unknown'
+    if (methodName === 'main') return projectStore.displayName
```

Comment:
If user has a fancy of naming some internal method `main` we would display project name instead. We should instead check if stack item is indeed a root item.